### PR TITLE
feat(app): resolve single slot location conflict

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

adds the necessary processing to useDeckConfigurationCompatibility to surface a required protocol single slot fixture conflict and provide the missing on-deck labware display name for that single slot. updates ODD FixtureTable to fix a location conflict modal bug where incorrect data was shown in the modal when multiple location conflicts exist

closes RAUT-885, RQA-1985

Screen Shot 2023-12-11 at 10 39 50 AM

# Test Plan

 - manually verified single slot conflict behavior
 - update unit tests

# Changelog

 -  Resolves single slot location conflict in protocol setup

# Review requests

 - load a protocol that requires a single slot labware in column 1/3
 - generate a conflict with a fixture by changing deck config to have a trash bin in that location
 - fixture list should contain an entry for the incorrect fixture (trash bin)
 - location conflict modal should say that protocol specifies the labware name expected in that slot

 - and check for other regressions

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/ProtocolSetupModulesAndDeck/FixtureTable.tsx [hunk 1]
```diff
@@ -89,98 +77,129 @@ export function FixtureTable({
...
+    showLocationConflictModal,
+    setShowLocationConflictModal,
+  ] = React.useState<boolean>(false)
+
+  const isCurrentFixtureCompatible =
+    cutoutFixtureId != null &&
+    compatibleCutoutFixtureIds.includes(cutoutFixtureId)
+  const isRequiredSingleSlotMissing = missingLabwareDisplayName != null
+  let chipLabel: JSX.Element
+  let handleClick
```
Comment (jerader) [inline app/src/organisms/ProtocolSetupModulesAndDeck/FixtureTable.tsx:125]: maybe this `handleClick` can be typed `() => void`

File: app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/LocationConflictModal.tsx [hunk 2]
```diff
@@ -44,6 +44,7 @@ import type {
 interface LocationConflictModalProps {
   onCloseClick: () => void
   cutoutId: CutoutId
+  missingLabwareDisplayName?: string | null
```
Comment (jerader) [inline app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/LocationConflictModal.tsx:47]: does this need to be typed as both null and undefined?

**Review by jerader — APPROVED** [review]
makes sense to me, tested in the dev env and works as expected. I left a few minor clean up questions

**Review by b-cooper — APPROVED** [review]
🌊

**Pull request merged**